src/OnlineRetailPortal.Web/Translators/AddProductRequestTranslator.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OnlineRetailPortal.Web.Translators
+{
+    public static class AddProductRequestTranslator
+    {
+        public static Contracts.AddProductRequest ToDataContract(this AddProductRequest addProductRequest)
+        {
+            Contracts.AddProductRequest request = new Contracts.AddProductRequest()
+            {
+                SellerId = addProductRequest.SellerId,
+                Name = addProductRequest.Name,
+                Description = addProductRequest.Description,
+                HeroImage = new Contracts.Image { Url = addProductRequest.HeroImage.Url },
+                Price = new Contracts.Price { Value = new Contracts.Value(addProductRequest.Price.Value.Amount, addProductRequest.Price.Value.Currency), IsNegotiable = addProductRequest.Price.IsNegotiable },
+                Category = (Contracts.Category)addProductRequest.Category,
```

Comment:
Use translators with ToEntity/ToModel instead of casting.

Reply:
Comment applies throughtout.

Reply:
done